NitroxClient/GameLogic/Bases/GhostMetadataApplier.cs

```diff
@@ -0,0 +1,209 @@
+using NitroxClient.Unity.Helper;
+using NitroxModel.DataStructures.GameLogic.Entities.Metadata;
+using NitroxModel.DataStructures.GameLogic.Entities.Metadata.Bases;
+using NitroxModel_Subnautica.DataStructures;
+using System.Collections;
+using UnityEngine;
+using UWE;
+
+namespace NitroxClient.GameLogic.Bases;
+
+public static class GhostMetadataApplier
+{
+    public static T From<T>(BaseGhost baseGhost) where T : GhostMetadata, new()
+    {
+        T metadata = new()
+        {
+            TargetOffset = baseGhost.targetOffset.ToDto()
+        };
+        return metadata;
+    }
```

Comment:
I find this kind of unintuitive. What about moving this to a constructor in GhostMetadata? I remember talking in Discord about this but can't remember the conclusion we had (if any).